pulsar-broker/src/main/java/com/yahoo/pulsar/broker/service/persistent/PersistentDispatcherMultipleConsumers.java

```diff
@@ -94,6 +94,8 @@ public PersistentDispatcherMultipleConsumers(PersistentTopic topic, ManagedCurso
         this.readBatchSize = MaxReadBatchSize;
         this.maxUnackedMessages = topic.getBrokerService().pulsar().getConfiguration()
                 .getMaxUnackedMessagesPerSubscription();
+        this.cursor.getNotDeletedMessages().forEach(position -> this.messagesToReplay
```

Comment:
Can we change it to getNotDeletedMessages(this.messagesToReplay) to reduce garbage due to PositionImpl

 i.e directly populate messagesToReplay instead of creating a temporary Set<PositionImpl> which will eventually be GC'ed.